_posts/2024-10-02-debug-with-maven-wildfly-plugin.adoc

```diff
@@ -0,0 +1,198 @@
+----
+<plugin>
+    <groupId>org.wildfly.plugins</groupId>
+    <artifactId>wildfly-maven-plugin</artifactId>
+    <version>${version.maven.wildfly.plugin}</version>
+    <configuration>
+        <galleon-options>
+            <jboss-fork-embedded>true</jboss-fork-embedded>
+        </galleon-options>
+    </configuration>
+    <executions>
+        <execution>
+            <id>server-provisioning</id>
+            <phase>generate-test-resources</phase>
+            <goals>
+                <goal>provision</goal>
+            </goals>
+            <configuration>
+                <provisioning-dir>${jboss.home}</provisioning-dir>
+                <galleon-options>
+                    <jboss-fork-embedded>${galleon.fork.embedded}</jboss-fork-embedded>
+                </galleon-options>
+                <feature-packs>
+                    <feature-pack>
+                        <groupId>${server.test.feature.pack.groupId}</groupId>
+                        <artifactId>${server.test.feature.pack.artifactId}</artifactId>
+                        <version>${wildfly.version}</version>
+                        <inherit-configs>false</inherit-configs>
+                        <included-configs>
+                            <config>
+                                <model>standalone</model>
+                                <name>standalone-full.xml</name>
+                            </config>
+                            <config>
+                                <model>standalone</model>
+                                <name>standalone.xml</name>
+                            </config>
+                        </included-configs>
+                        <excluded-packages>
+                            <name>docs.schema</name>
+                            <name>appclient</name>
+                            <name>domain</name>
+                        </excluded-packages>
+                    </feature-pack>
+                </feature-packs>
+                <channels>
+                    <channel>
+                        <manifest>
+                            <groupId>org.jberet</groupId>
+                            <artifactId>jberet-channel-manifest</artifactId>
+                            <version>${version.jberet}</version>
+                        </manifest>
+                    </channel>
+                </channels>
+            </configuration>
+        </execution>
+        ...
+    </executions>
+</plugin>
+----
+
+In the above configuration, the `provision` goal is configured to run in the `generate-test-resources` phase. In this execution, the `feature-packs` section is used to override the feature packs to be used by the provisioned server, and the `channels` section is using the `jberet-channel-manifest` channel to override the `batch-jberet` module to be used in the provisioned server. Please note that you need to use the `channels` with the `feature-packs` together. Here are the properties used by the configuration:
```

Comment:
s/is used to override/specifies/g

There is no overriding here, as there is no default feature pack that would be used if this wasn't set.